pulley/src/interp/tail_loop.rs

```diff
@@ -59,20 +64,47 @@ macro_rules! tail_call {
 }
 
 impl Interpreter<'_> {
-    pub fn run(mut self) -> Done {
-        // Perform a dynamic dispatch through a function pointer indexed by
-        // opcode.
-        let opcode = unwrap_uninhabited(Opcode::decode(self.bytecode()));
-        let handler = OPCODE_HANDLER_TABLE[opcode as usize];
-        tail_call!(handler(self));
+    pub fn run(self) -> Done {
+        dispatch(self.state, self.pc, self.executing_pc)
     }
 }
 
+fn debug<'a>(
+    state: &'a mut MachineState,
+    pc: UnsafeBytecodeStream,
+    executing_pc: ExecutingPcRef<'a>,
+) -> debug::Debug<'a> {
+    debug::Debug(Interpreter {
+        state,
+        pc,
+        executing_pc,
+    })
+}
+
+fn dispatch(
+    state: &mut MachineState,
+    pc: UnsafeBytecodeStream,
+    executing_pc: ExecutingPcRef<'_>,
+) -> Done {
+    // Perform a dynamic dispatch through a function pointer indexed by
+    // opcode.
+    let mut debug = debug(state, pc, executing_pc);
+    debug.before_visit();
```

Comment:
I'll note here that we were forgetting this before, which caused the profiler to not work with the tail loop initially.